### PR TITLE
Developer Documentation for ID Repo

### PR DESCRIPTION
**Pull request by vipuldhurve02**

### CODE REVIEW

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# ID Repository Credential Request Generator Service Developers' Guide
+
+## Overview
+
+[ID Repository](https://docs.mosip.io/1.2.0/modules/id-repository) contains the records of identity of an individual, and provides API based mechanism to store, retrieve and update identity details by other MOSIP modules. ID Repository is used by Registration Processor, ID Authentication and Resident Services
```
Comment (manojsp12) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: update intro

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+5. Postman
+6. Git
+7. Notepad++ (optional)
+8. lombok.jar (file)
+9. settings.xml (document)
+
+
+Follow the steps below to set up ID Repository Services on your local system:
+
+1. Download `lombok.jar` and `settings.xml` from [here](https://github.com/mosip/documentation/tree/1.2.0/docs/_files/resident-services-config-files).
```
Comment (manojsp12) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: change url to idrepo config files

File: docs/_files/id-repository-config-files/settings,xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<settings>
```
Comment (manojsp12) [inline docs/_files/id-repository-config-files/settings,xml]: update file name to settings.xml

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+ 
+## Code setup
+
+For the code setup, clone the repository and follow the guidelines mentioned in the [Code Contributions](https://docs.mosip.io/1.2.0/community/code-contributions).
+
+### Importing and building of a project
+
+1. Open the project folder where `pom.xml` is present.
+1. Open command prompt from the same folder.
+1. Run the command `mvn clean install -Dgpg.skip=true -DskipTests=true` to build the project and wait for the build to complete successfully.
```
Comment (manojsp12) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: remove -DskipTests=true

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+1. For the environment setup, you need an external JAR that is available [here](https://oss.sonatype.org/#nexus-search;gav~~kernel-auth-adapter~1.2.0-SNAPSHOT~~) with different versions. (E.g.: You can download `kernel-auth-adapter.jar` and add to project `Libraries → Classpath → Add External JARs → Select Downloaded JAR → Add → Apply and Close`).
+
+<img src="_images/add-external-library.png" width="750" height="450">
+
+2. Clone [mosip-config repository](https://github.com/mosip/mosip-config).
+
+3. Create an empty folder inside the `mosip-config` with `sandbox-local` name and then copy and paste all config files inside `sandbox-local` folder except `.gitignore, README and LICENSE`.
+
+4. As Id Repository is using two properties files, `id-repository-default` and `application-default`, you will have to configure them according to your environment.    The same files are available [here](https://github.com/mosip/documentation/tree/1.2.0/docs/_files/resident-services-config-files) for reference.
```
Comment (manojsp12) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: update url to idrepo

Comment (manojsp12) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: Properties to be updated:
application-default.properties
1. property1=<configure password>
2. propety2 = value

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+4. Click Apply and then debug it (starts running).
+
+## Credential Request Generator Service API
+​
+* For API documentation, refer [here](https://docs.mosip.io/1.2.0/api).
+​
+* The APIs can be tested with the help of **Swagger-UI**. 
+​
+* Swagger is an interface description language for describing restful APIs expressed using JSON. You can access Swagger-UI of credential-request-generator-services for dev-environment from `https://localhost:8094/v1/credentialrequest/swagger-ui/index.html?configUrl=/v1/credentialrequest/v3/api-docs/swagger-config#/` and localhost from `https://localhost:8094/v1/credentialrequest/swagger-ui/index.html?configUrl=/v1/credentialrequest/v3/api-docs/swagger-config#/`.
```
Comment (manojsp12) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: dev-environment not required

**Review by manojsp12 — CHANGES_REQUESTED** [review]

File: docs/_files/id-repository-config-files/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# ID REPOSITORY SERVICES CONFIGURATION FILES
+
+## Overview
+
+This folder contains the files required for local configurations of Resident services.
```
Comment (SwatiPeriwal) [inline docs/_files/id-repository-config-files/README.md]: This folder contains the files required for local configurations of Resident services.    --- why Resident services?

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+1. Key Manager encrypts/decrypts data.
+2. Auth Adapter integrates with KeyCloak for authentication.
+
+The documentation here will guide you through the prerequisites required for the developer' setup.
+
+
+## Software setup
+
+Below are a list of tools required in ID Repoistory Services:
```
Comment (SwatiPeriwal) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: Repository- spell check

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+1. For the environment setup, you need an external JAR that is available [here](https://oss.sonatype.org/#nexus-search;gav~~kernel-auth-adapter~1.2.0-SNAPSHOT~~) with different versions. (E.g.: You can download `kernel-auth-adapter.jar` and add to project `Libraries → Classpath → Add External JARs → Select Downloaded JAR → Add → Apply and Close`).
+
+<img src="_images/add-external-library.png" width="750" height="450">
+
+2. Clone [mosip-config repository](https://github.com/mosip/mosip-config).
+
+3. Create an empty folder inside the `mosip-config` with `sandbox-local` name and then copy and paste all config files inside `sandbox-local` folder except `.gitignore, README and LICENSE`.
+
+4. As Id Repository is using two properties files, `id-repository-default` and `application-default`, you will have to configure them according to your environment.    The same files are available [here](https://github.com/mosip/documentation/tree/1.2.0/docs/_files/id-repository-config-files) for reference.
```
Comment (SwatiPeriwal) [inline docs/id-repository-credential-request-generator-service-developer-guide.md]: ID -capitalization

File: docs/id-repository-credential-request-generator-service-developer-guide.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+5. To run the server, two files are required- [kernel-config-server.jar](https://oss.sonatype.org/#nexus-search;gav~~kernel-config-server~1.2.0-SNAPSHOT~~) and [config-server-start.bat](https://github.com/mosip/documentation/blob/1.2.0/docs/_files/id-repository-config-files/config-server-start.bat).
+
+6. Put both the files in the same folder and change the location attribute to `sandbox-local` folder in `config-server-start.bat` file and also check the version of `kernel-config-server.jar` towards the end of the command. 
+
+ Example:
+ 
+  ```java -jar -Dspring.profiles.active=native  -Dspring.cloud.config.server.native.search-locations=file:C:\Users\myDell\mosipProject\mosip-config\sandbox-local -Dspring.cloud.config.server.accept-empty=true  -Dspring.cloud.config.server.git.force-pull=false -Dspring.cloud.config.server.git.cloneOnStart=false -Dspring.cloud.config.server.git.refreshRate=0 kernel-config-server-1.2.0-20201016.134941-57.jar```.
+
+7. Run the server by opening the `config-server-start.bat` file.
```
Comment (SwatiPeriwal) [inline docs/id-repository-credential-request-generator-service-developer-guide.md:94]: The entire content in in blue..pls check code block closure

File: docs/id-repository-credential-service-developer-guide.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+<img src="_images/id-repository-credential-service.png" width="750" height="450">
+
+1. Key Manager encrypts/decrypts data and also used to sign data.
+2. WebSub subscribes to get notifications related to credential status from IDA.
+3. [DataShare](https://docs.mosip.io/1.2.0/modules/datashare) creates datashare url for sharable attributes.
+4. Identity service retrieves identity data for UIN/VID.
+5. artner management service retrieves policies related to credential type and also retrieves policy for bio-extraction.
+6. Auth Adapter integrates with KeyCloak for authentication.
+
+### Credential Types
```
Comment (SwatiPeriwal) [inline docs/id-repository-credential-service-developer-guide.md]: Repeated content

**Review by SwatiPeriwal — CHANGES_REQUESTED** [review]
Please check the changes and revert

File: docs/id-repository-credential-request-generator-service-developer-guide.md (hunk 9 above)
Comment (vipuldhurve02) [reply docs/id-repository-credential-request-generator-service-developer-guide.md]: It is exactly mentioned as per the name of the "id-repository-default.properties" file.

**Comment by manojsp12** [comment]
@vipuldhurve02 I see a commit [Delete id-repository-credential-service-developer-guide.md](https://github.com/mosip/documentation/pull/199/commits/e3223cdb6bd7ac18083dc6c9c7f39bde72626ded). why credential service doc got deleted?

**Comment by vipuldhurve02** [comment]
> @vipuldhurve02 I see a commit [Delete id-repository-credential-service-developer-guide.md](https://github.com/mosip/documentation/pull/199/commits/e3223cdb6bd7ac18083dc6c9c7f39bde72626ded). why credential service doc got deleted?

**Comment by SwatiPeriwal** [comment]
Approving..

**Pull request merged**